gooresmed/download.py

```diff
@@ -124,6 +124,41 @@ class Download(_DownloadBase):
            ``start`` to the end of the media.
     """
 
+    def _prepare_request(self):
+        """Prepare the contents of an HTTP request.
+
+        .. _sans-I/O: https://sans-io.readthedocs.io/
+
+        This is everything that must be done before a request that doesn't
+        require network I/O (or other I/O). This is based on the `sans-I/O`_
+        philosophy.
+
+        Returns:
+            dict: The headers for the request.
+
+        Raises:
+            ValueError: If the current :class:`Download` has already
+                finished.
+        """
+        if self.finished:
+            raise ValueError('A download can only be used once.')
+
+        headers = {}
+        _add_bytes_range(self.start, self.end, headers)
+        return headers
+
+    def _process_response(self):
```

Comment:
One thing you might think about is an abc for "HTTP Pipeline": https://github.com/haikuginger/httppipeline/blob/master/httppipeline/sample.py